src/components/brightness/BrightnessController.h

```diff
@@ -20,6 +24,17 @@ namespace Pinetime {
 
     private:
       Levels level = Levels::High;
+      static constexpr uint8_t UNSET = UINT8_MAX;
+      uint8_t lastPin = UNSET;
+      nrfx_timer_t timer;
+      // warning: nimble reserves some ppis
+      // https://github.com/InfiniTimeOrg/InfiniTime/blob/034d83fe6baf1ab3875a34f8cee387e24410a824/src/libs/mynewt-nimble/nimble/drivers/nrf52/src/ble_phy.c#L53
+      // spimaster uses ppi 0 for an erratum workaround
+      // channel 1, 2 should be free to use
+      nrf_ppi_channel_t ppi_backlight_on = NRF_PPI_CHANNEL1;
```

Comment:
Let's try from my and the [internets](https://en.cppreference.com/w/cpp/language/constexpr) understanding:

`static` means that this will not be bound to a specific instance of the class. Sounds good for a general pin config.

`constexpr` let the compiler evaluate this at compile-time, so there is no runtime overhead in the resulting image. In classes, this also implies `const`, meaning that it cannot be changed later at runtime. And in `static` declarations it even implies `inline`, so the value will be copied everywhere at compile time instead of loading it from the variable. 

`static constexpr` is fine here, if this value will never change and you want to use it everywhere with as less runtime overhead as possible.

I don't know exactly what the code does, but that seems to be the case here.

Reply:
Thanks for the explainer, looks like static constexpr is appropriate here then :+1:

In this case the ppi_backlight_on/off are constants designating the channel to be used for the backlight on/off triggers